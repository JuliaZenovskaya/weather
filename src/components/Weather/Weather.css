.weather {
    margin: 30px;
    font-size: 20px;
    border: 2px green ridge;
}

.left {
  float: left;
  margin-left: 10px;
}
