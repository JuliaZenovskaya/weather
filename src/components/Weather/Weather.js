import React from "react";
import './Weather.css';

class WeatherData extends React.Component {
  componentDidMount() {
    if (this.props.getWeather) this.props.getWeather();
  }

  render() {
    if (!this.props.weather) {
        return <div>Загрузка...</div>
    }

    return (
      <div className="weather">
      <div className="left">
        <div>{this.props.weather.name}</div>
        <img src={`https://openweathermap.org/img/wn/${this.props.weather.weather[0].icon}.png`} />
        <div>{Math.round(this.props.weather.main.temp)-273} °C</div>
        {this.props.onDelete && <button className="button" onClick={this.props.onDelete}>Удалить</button>}
      </div>
      <div className="right">
        <div className="pressure">Давление: {this.props.weather.main.pressure} hPa</div>
        <div className="humidity">Влажность: {this.props.weather.main.humidity}%</div>
        <div className="clouds">Облачность: {this.props.weather.weather[0].description}</div>
        <div className="wind">Ветер: {this.props.weather.wind.speed} м/с</div>
        <div className="coords">Координаты: [{this.props.weather.coord.lon}, {this.props.weather.coord.lat}]</div>
      </div>
      </div>
    );
  }
}

export default WeatherData
