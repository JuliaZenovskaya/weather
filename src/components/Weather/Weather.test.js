import React from "react";
import Weather from "./Weather";
import renderer from "react-test-renderer";

const response = {
    "coord":{"lon":37.62,"lat":55.75},
    "weather":[{"id":701,"main":"Mist","description":"mist","icon":"50n"}],
    "base":"stations",
    "main":{"temp":4.38,"pressure":1015,"humidity":100,"temp_min":2,"temp_max":5.56},
    "visibility":3000,
    "wind":{"speed":4,"deg":90},
    "clouds":{"all":90},
    "dt":1573170827,
    "sys":{"type":1,"id":9027,"country":"RU","sunrise":1573188558,"sunset":1573220232},
    "timezone":10800,
    "id":524901,
    "name":"Moscow",
    "cod":200
  };

describe("Weather for geolocation", () => {
  it ("Weather has loaded", () => {
    const htmlTree = renderer.create(<Weather weather={response}/>).toJSON();
    expect(htmlTree).toMatchSnapshot();
  });
  it ("Weather is loading", () => {
    const htmlTree = renderer.create(<Weather/>).toJSON();
    expect(htmlTree).toMatchSnapshot();
  });
});

describe("Weather for the city", () => {
  it ("Weather has loaded", () => {
    const htmlTree = renderer.create(<Weather
          getWeather={()=> {}}
          onDelete={()=> {}}/>).toJSON();
    expect(htmlTree).toMatchSnapshot();
  });
  it ("Weather is loading", () => {
    const htmlTree = renderer.create(<Weather
          weather={response}
          getWeather={()=> {}}
          onDelete={()=> {}}/>).toJSON();
    expect(htmlTree).toMatchSnapshot();
  });
});
