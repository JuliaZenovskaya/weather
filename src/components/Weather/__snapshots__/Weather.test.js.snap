// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather for geolocation Weather has loaded 1`] = `
<div
  className="weather"
>
  <div
    className="left"
  >
    <div>
      Moscow
    </div>
    <img
      src="https://openweathermap.org/img/wn/50n.png"
    />
    <div>
      -269
       °C
    </div>
  </div>
  <div
    className="right"
  >
    <div
      className="pressure"
    >
      Давление: 
      1015
       hPa
    </div>
    <div
      className="humidity"
    >
      Влажность: 
      100
      %
    </div>
    <div
      className="clouds"
    >
      Облачность: 
      mist
    </div>
    <div
      className="wind"
    >
      Ветер: 
      4
       м/с
    </div>
    <div
      className="coords"
    >
      Координаты: [
      37.62
      , 
      55.75
      ]
    </div>
  </div>
</div>
`;

exports[`Weather for geolocation Weather is loading 1`] = `
<div>
  Загрузка...
</div>
`;

exports[`Weather for the city Weather has loaded 1`] = `
<div>
  Загрузка...
</div>
`;

exports[`Weather for the city Weather is loading 1`] = `
<div
  className="weather"
>
  <div
    className="left"
  >
    <div>
      Moscow
    </div>
    <img
      src="https://openweathermap.org/img/wn/50n.png"
    />
    <div>
      -269
       °C
    </div>
    <button
      className="button"
      onClick={[Function]}
    >
      Удалить
    </button>
  </div>
  <div
    className="right"
  >
    <div
      className="pressure"
    >
      Давление: 
      1015
       hPa
    </div>
    <div
      className="humidity"
    >
      Влажность: 
      100
      %
    </div>
    <div
      className="clouds"
    >
      Облачность: 
      mist
    </div>
    <div
      className="wind"
    >
      Ветер: 
      4
       м/с
    </div>
    <div
      className="coords"
    >
      Координаты: [
      37.62
      , 
      55.75
      ]
    </div>
  </div>
</div>
`;
